Triviagen-7: Create main menu screen and change theme

Kazoroo opened the pull request:
Main menu screen
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/ff8db9ad-e620-4c78-8fb7-3faaac6fbe74)

App logo
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/09794062-4aba-45b2-be06-e5d6e8379e65)

I have created a package presentation in the core, where I put TriviaGenScaffold, so I can be reusable not only in trivia feature. Also changes to Scaffold like adding bottom bar are applied to each screen that uses this function.

Commit:
Add new launcher icon

Commit:
Add app primary color, change typography, set app background color to brand color.

Took 2 hours 34 minutes

Commit:
Add main menu screen. Create reusable Scaffold function.

Took 10 minutes

sidcgithub commented on app/src/main/ic_launcher-playstore.png:
Do we need this png here? Maybe we can just keep the webp versions.

Kazoroo replied:
Right, I haven't noticied it.

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/MainActivity.kt:
```diff
@@ -1,49 +1,31 @@
-                        name = "Android",
-                        modifier = Modifier.padding(innerPadding)
-                    )
+                Scaffold(
```
`MainMenuScreen` has a `Scaffold`. Do we need an external `Scaffold` too?

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/trivia/presentation/mainmenu/MainMenuScreen.kt:
```diff
@@ -0,0 +1,60 @@
+                    .size(200.dp)
+            )
+            
+            Spacer(modifier = Modifier.height(50.dp))
```
I will possibly have to update the dimensions in Figma and make them uniform but can we use dimension resources?
https://developer.android.com/develop/ui/compose/resources#dimensions

sidcgithub commented on app/src/main/res/drawable-nodpi/yellow_app_logo.png:
nit: If you preview the Figma file this is an animation. Not necessary but if you'd like we could have an animation effect. The Gif is blurry but maybe you can use this to switch between black and yellow colour filters -> https://developer.android.com/develop/ui/compose/animation/vectors

![flicker](https://github.com/sidcgithub/ai-trivia-app-android/assets/12014497/69a3a6cc-90c1-4424-ab66-2c8732541db0)

sidcgithub replied:
This may be a better resource for animations. I believe value based animations will likely be helpful:
https://developer.android.com/develop/ui/compose/animation/introduction#value-based-animations

Kazoroo replied:
The animation is not good quality, I wouldn't implement that. In my opinion it will look terrible with those black dots.

sidcgithub replied:
Ok. I agree not the best animation

sidcgithub reviewed (commented):
Hey this is great! Just a few comments.

Commit:
Fixes and improvments

Remove dynamic color, delete unused png, delete redundant Scaffold from MainActivity.kt, add dimens.xml and use those values in MainMenuScreen.

Took 24 minutes

Commit:
Optimize imports

Took 41 seconds

sidcgithub reviewed (approved):
LGTM. Any thoughts on using a fork vs branches for Pull Requests? Any pros and cons?

Kazoroo commented:
> LGTM. Any thoughts on using a fork vs branches for Pull Requests? Any pros and cons?

I think the fork is better, I can work in local environment, don't affecting anything.
